lib/http-bridge/test/integration/Test/Integration/Scenario/CLISpec.hs

```diff
@@ -0,0 +1,110 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Test.Integration.Scenario.CLISpec
+    ( specNoCluster
+    , specWithCluster
+    ) where
+
+import Prelude
+
+import Cardano.Wallet.Api.Types
+    ( ApiWallet )
+import Control.Monad
+    ( forM_ )
+import Data.List
+    ( length )
+import Data.List.Split
+    ( splitOn )
+import Data.Text
+    ( Text )
+import System.Command
+    ( Stdout (..), command )
+import Test.Hspec
+    ( SpecWith, describe, it )
+import Test.Hspec.Expectations.Lifted
+    ( shouldBe, shouldContain, shouldNotContain )
+import Test.Integration.Framework.DSL
+    ( Context (..)
+    , Headers (..)
+    , Payload (..)
+    , expectResponseCode
+    , getFromResponse
+    , json
+    , request
+    , walletId
+    )
+
+import qualified Data.Text as T
+import qualified Network.HTTP.Types.Status as HTTP
+
+specNoCluster :: SpecWith ()
+specNoCluster = do
+
+    it "CLI - Shows version" $  do
+        Stdout out <- command [] "cardano-wallet" ["--version"]
+        cabal <- readFile "../../cardano-wallet.cabal"
+        cabal `shouldContain` out
+
+    it "CLI - Can generate mnemonics with default size" $  do
+        Stdout out <- command [] "cardano-wallet" ["mnemonic", "generate"]
+        length (splitOn " " out) `shouldBe` 15
+
+    describe "CLI - Can generate mnemonics with different sizes" $  do
+        forM_ [9, 12, 15, 18, 21, 24] $ \size -> it (show size) $ do
+            Stdout out <- command [] "cardano-wallet"
+                            ["mnemonic", "generate", "--size", show size]
+            length (splitOn " " out) `shouldBe` size
+
+specWithCluster :: SpecWith Context
+specWithCluster = do
+
+    it "CLI - Can get a wallet" $ \ctx -> do
+        walId <- createWallet ctx "1st CLI Wallet" mnemonics15
+        Stdout out <- command [] "cardano-wallet"
+                        ["wallet", "get", "--port", "1337", walId ]
+        out `shouldContain` "1st CLI Wallet"
+
+    it "CLI - Can list wallets" $ \ctx -> do
+        _ <- createWallet ctx "1st CLI Wallet" mnemonics15
+        _ <- createWallet ctx "2nd CLI Wallet" mnemonics18
+        Stdout out <- command [] "cardano-wallet"
+                        ["wallet", "list", "--port", "1337"]
+        out `shouldContain` "1st CLI Wallet"
+        out `shouldContain` "2nd CLI Wallet"
+
+    it "CLI - Can update wallet name" $ \ctx -> do
+        walId <- createWallet ctx "1st CLI Wallet" mnemonics15
+        Stdout out <- command [] "cardano-wallet"
+                        ["wallet", "update", "--port", "1337"
+                        , walId, "--name", "Updated name" ]
+        out `shouldContain` "Updated name"
+
+    it "CLI - Can delete wallet" $ \ctx -> do
+        walId <- createWallet ctx "CLI Wallet" mnemonics15
+        Stdout out <- command [] "cardano-wallet"
+                        ["wallet", "delete", "--port", "1337", walId ]
+        out `shouldNotContain` "CLI Wallet"
```

Comment:
Interesting approach. I think we could add some special assertion function too here to verify the output even more deeply. The CLI do output some `Ok` or `Error` on `stderr` and, if any, output responses on `stdout` (without any noise). So in practice, we could even verify that whatever gets printed on `stdout` is always a valid JSON object.

The `CmdResult` allows us to check both stderr and stdout, and also response codes! Which is kinda cool. So we can definitely go crazy here :stuck_out_tongue: 

Side-note: variable indentation --> not cool.

Reply:
Definitely there's room for going _crazy_ and I had the same _idea_ to check the somehow _cast_ the output to JSON or maybe even `ApiWallet` (still the main intention here was to check if the approach works)

>Side-note: variable indentation --> not cool.

it was not my intention :) it's just trying to comply with 80 chars rule...